app/containers/HomePage.jsx

```diff
@@ -6,16 +6,25 @@ import * as actions from '../redux/actions/index';
 import adb from '../utils/adb';
 import AppBar from 'material-ui/AppBar';
 import { push } from 'react-router-redux';
+import result from 'lodash/result';
 import HeaderIcons from '../components/Header/Buttons';
 
 class HomePage extends Component {
+  autoRefresh = () => {
+    setInterval(() => {
+      if (result(this.props.state, 'devices.devicesList', []).length === 0) {
+        this.listDevices();
+      }
+    }, 2000);
+  }
   listDevices() {
     const vm = this;
     adb.listDevices().then(devices => {
       vm.props.actions.updateDevices(devices);
       devices.forEach((eachDevice) => {
         adb.getDeviceProperties(eachDevice.serial).then((deviceProperties) => {
           vm.props.actions.updateDeviceProperties(eachDevice.serial, deviceProperties);
+          vm.autoRefresh();
```

Comment:
Better to call this in the constructor. 
Otherwise , its going to register a setInterval everytime we hit listDevices()